handlers/attachment_resource.go

```diff
@@ -159,3 +159,92 @@ func HandleGetAttachmentDetails(svc dao.Service) http.Handler {
 		utils.WriteJSONWithStatus(w, req, attachmentResponse, http.StatusOK)
 	})
 }
+
+// HandleDownloadAttachment download an attachment which is stored against an Insolvency case
+func HandleDownloadAttachment(svc dao.Service) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		vars := mux.Vars(req)
+		transactionID := utils.GetTransactionIDFromVars(vars)
+		if transactionID == "" {
+			log.ErrorR(req, fmt.Errorf("there is no transaction ID in the URL path"))
+			m := models.NewMessageResponse("transaction ID is not in the URL path")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusBadRequest)
+			return
+		}
+
+		attachmentID := utils.GetAttachmentIDFromVars(vars)
+		if attachmentID == "" {
+			log.ErrorR(req, fmt.Errorf("there is no attachment ID in the URL path"))
+			m := models.NewMessageResponse("attachment ID is not in the URL path")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusBadRequest)
+			return
+		}
+
+		// Get attachment data from DB to check if attachment ID is valid
+		attachmentResource, err := svc.GetAttachmentFromInsolvencyResource(transactionID, attachmentID)
```

Comment:
Should we be checking if the attachment ID is valid or not? I think the appropriate course of action here would be to just query the DB to see if an attachment resource is returned or not. If nil is returned, we should just be returning a `404 - Not Found` instead of a `400 - Bad Request`